src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs

```diff
@@ -19,7 +19,7 @@ public class CSharpImmutableObjectMethodAnalyzer : DiagnosticAnalyzer
         private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(CodeAnalysisDiagnosticsResources.DoNotIgnoreReturnValueOnImmutableObjectMethodInvocationMessage), CodeAnalysisDiagnosticsResources.ResourceManager, typeof(CodeAnalysisDiagnosticsResources));
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(CodeAnalysisDiagnosticsResources.DoNotIgnoreReturnValueOnImmutableObjectMethodInvocationDescription), CodeAnalysisDiagnosticsResources.ResourceManager, typeof(CodeAnalysisDiagnosticsResources));
 
-        public static readonly DiagnosticDescriptor DoNotIgnoreReturnValueDiagnosticRule = new DiagnosticDescriptor(
+        public static readonly DiagnosticDescriptor DoNotIgnoreReturnValueDiagnosticRule = new(
```

Comment:
Can you please target 2.9.x branch (let's confirm with @dotpaul first though)? Otherwise, we will likely run into merge conflicts in future.

Reply:
Also if we are fixing all these issues, can we also remove the .editorconfig entries which were downgrading them to suggestion/silent/none so future violations fails the build?

Reply:
If targeting the 2.9.x branch is practical, please do.

Reply:
See #4559 for applicable changes in branch `2.9.x`.